### PR TITLE
Task10

### PR DESCRIPTION
**Pull request by ATGE**
- Completed implementing the client_manager following the TTD approach
- Created an abstract class called DBConnector with abstract methods
- Renamed the current DBConnector where Redis logic is, to be DBConnectorRedis

### CODE REVIEW

File: tests/unit/test_db_connector.py [hunk 1]
```diff
@@ -35,7 +38,22 @@ def test_get_by_id_mock_as_context_manager(mocker):
     mocker.patch("redis.Redis.get").return_value = json.dumps(object_to_save)
 
     result = None
-    with DBConnector() as db_connector:
+    with DBConnectorRedis.instance() as db_connector:
         db_connector.save(id_test, object_to_save)
         result = db_connector.get_by_id(id_test)
     assert result
```
Comment (favipcj) [inline tests/unit/test_db_connector.py:44]: here it was missing the comparison, which I forgot to upload fyi

File: truck_delivery_atge/client.py [hunk 2]
```diff
@@ -3,9 +3,10 @@
 
 class Client(Person):
     """ Class representing a Client """
-    def __init__(self, name, email, cellphone, address, nit):
-        super(Client, self).__init__(name, email, cellphone, address)
+    def __init__(self,ci='', name='', email='', cellphone='', address = '', nit='', contract_number=''):
```
Comment (favipcj) [inline truck_delivery_atge/client.py:6]: as we are going to use the ci as identifier, that parameter should not be optional

File: truck_delivery_atge/client_manager.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+
+from truck_delivery_atge.content_manager import ContentManager
+from truck_delivery_atge.db_connector import DBConnector
+from truck_delivery_atge.client import Client
+
+CLIENTE_ID = "UUI-client"
```
Comment (favipcj) [inline truck_delivery_atge/client_manager.py:6]: you should be importing the UUI from the utils module

**Review by favipcj — APPROVED** [review]

File: truck_delivery_atge/db_connector_redis.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self):
+         super(DBConnectorRedis, self).__init__()
+
+    @classmethod
+    def instance(cls):
+        if cls._instance is None:
+            cls._instance = cls.__new__(cls)
+            cls.connection = redis.Redis(host="localhost",
+                                          port=6379)
+        return cls._instance
```
Comment (favipcj) [inline truck_delivery_atge/db_connector_redis.py:21]: Nice work doing this, you can check this out as python have dunder methods that can be overwritten to achieve the same result but in a more python way, as it's mentioned here https://python-patterns.guide/gang-of-four/singleton/#a-more-pythonic-implementation

File: truck_delivery_atge/client.py (hunk 2 above)
Comment (ATGE) [reply truck_delivery_atge/client.py:6]: updated

File: tests/unit/test_db_connector.py (hunk 1 above)
Comment (ATGE) [reply tests/unit/test_db_connector.py:44]: comparison added

File: truck_delivery_atge/client_manager.py (hunk 3 above)
Comment (ATGE) [reply truck_delivery_atge/client_manager.py:6]: added

**Review by ATGE — COMMENTED** [review]
@favipcj suggestions added

**Pull request merged**